lib/asset-manifest-inserter.js

```diff
@@ -55,16 +62,31 @@ AssetManifestInserter.prototype.build = function() {
   if (existsSync(indexFilePath)) {
     var indexFile = fs.readFileSync(indexFilePath, { encoding: 'utf8' });
     var index = this.replacer(indexFile, appTransformedManifest);
-    fs.writeFileSync(path.join(this.outputPath, this.options.indexName), index);
+
+    // If index and lastIndex match, make this a noop to avoid triggering full page refresh
+    if (index !== this.lastIndex) {
+      this.lastIndex = index;
+      fs.writeFileSync(path.join(this.outputPath, this.options.indexName), index);
+    }
+
   }
 
   var testsTransformedManifest = this.transformer(manifest, 'test');
+  var testDirectory = path.join(this.outputPath, 'tests');
+
+  if (!existsSync(testDirectory)) {
+    fs.mkdirSync(path.join(this.outputPath, 'tests'));
```

Comment:
I think this should be moved down into the `if (testIndex !== this.lastTestIndex) {` conditional below, does that work?

Reply:
Done